esx_service/utils/auth_api.py

```diff
@@ -241,7 +247,12 @@ def get_default_datastore(name):
 
 def _tenant_create(name, description="", vm_list=None, privileges=None):
     """ API to create a tenant """
+    global valid_tenant_name_reg
     logging.debug("_tenant_create: name=%s description=%s vm_list=%s privileges=%s", name, description, vm_list, privileges)
+    if not valid_tenant_name_reg.match(name):
+        error_info = error_code.generate_error_info(ErrorCode.TENANT_NAME_INVALID, name, VALID_TENANT_NAME_REGEXP)
+        return error_info, None
+
```

Comment:
Can this be a function like is_tenant_name_valid() vs. repeated in multiple locations.